vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -0,0 +1,346 @@
+        "marks, expected",
+        [
+            ({i: str(i) for i in range(0, 10, 5)}, {i: str(i) for i in range(0, 10, 5)}),
+            ({15: 15, 25: 25}, {15.0: "15", 25.0: "25"}),
+            ({"15": 15, "25": 25}, {15.0: "15", 25.0: "25"}),
+            (None, None),
+        ],
+    )
+    def test_valid_marks(self, marks, expected):
+        range_slider = vm.RangeSlider(min=0, max=10, marks=marks)
+
+        assert range_slider.marks == expected
+
+    def test_invalid_marks(self):
+        with pytest.raises(
+            ValidationError,
+            match="2 validation errors for RangeSlider",
+        ):
+            vm.RangeSlider(min=1, max=10, marks={"start": 0, "end": 10})
+
+    @pytest.mark.parametrize("step, expected", [(1, {}), (None, None)])
+    def test_set_default_marks(self, step, expected):
+        slider = vm.RangeSlider(min=0, max=10, step=step)
+        assert slider.marks == expected
+
+    @pytest.mark.parametrize(
+        "title",
+        [
+            "test",
+            1,
+            1.0,
+            """## Test header""",
+            "",
+        ],
+    )
+    def test_valid_title(self, title):
+        slider = vm.RangeSlider(title=title)
+
+        assert slider.title == str(title)
+
+    def test_set_action_via_validator(self, test_action_function):
+        range_slider = vm.RangeSlider(actions=[vm.Action(function=test_action_function)])
+        actions_chain = range_slider.actions[0]
+
+        assert actions_chain.trigger.component_property == "value"
+
+
+class TestRangeSliderBuild:
+    """Tests model build method."""
+
+    def test_range_slider_build_default(self, expected_range_slider_default):
+        range_slider = vm.RangeSlider(id="range_slider")
+        component = range_slider.build()
+
+        result = json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_range_slider_default, cls=plotly.utils.PlotlyJSONEncoder))
+
+        assert result == expected
+
+    def test_range_slider_build_with_optional(self, expected_range_slider_with_optional):
+        range_slider = vm.RangeSlider(min=0, max=10, step=1, value=[0, 10], id="range_slider_with_all", title="Title")
+        component = range_slider.build()
+
+        result = json.loads(json.dumps(component, cls=plotly.utils.PlotlyJSONEncoder))
+        expected = json.loads(json.dumps(expected_range_slider_with_optional, cls=plotly.utils.PlotlyJSONEncoder))
+
+        assert result == expected
+
+
+class TestCallbackMethod:
+    """Tests model callback method."""
+
+    @pytest.mark.parametrize(
+        "trigger_id, start, end, slider_value, input_store, value, expected_value",
+        [
+            (
+                "_start_value",
+                2,
+                10,
+                [3, 10],
+                [1, 10],
+                None,
+                (2, 10, [2, 10], (2, 10)),
+            ),  # change start value by left handle
```

Comment:
```suggestion
            (  # user changes left text field, slider and store update
                "_start_value",
                2,
                10,
                [3, 10],
                [1, 10],
                None,
                (2, 10, [2, 10], (2, 10)),
            ),
```
Could we try to do it in the above style, I still don't find it clear enough. Part of that has to do with maybe renaming start to start_txt or so (if that's correct).

In this example, are [3,10] and [1,10] correct? Why would they ever be different